packages/twenty-front/src/modules/object-record/record-table/hooks/useSetTableColumns.ts

```diff
@@ -0,0 +1,33 @@
+import { FieldMetadata } from '@/object-record/record-field/types/FieldMetadata';
+import { tableColumnsComponentState } from '@/object-record/record-table/states/tableColumnsComponentState';
+import { ColumnDefinition } from '@/object-record/record-table/types/ColumnDefinition';
+import { getSnapshotValue } from '@/ui/utilities/state/utils/getSnapshotValue';
+import { useRecoilCallback } from 'recoil';
+import { isDeeplyEqual } from '~/utils/isDeeplyEqual';
+
+export const useSetTableColumns = () => {
+  const setTableColumns = useRecoilCallback(
```

Comment:
This should be refactored after advanced filters, good point for extracting in a separate hook.